Automattic/jetpack
# In Progress: Remove post limit on sitemaps

Addresses #3314: Remove 1000 post limit on sitemaps.
#### Changes proposed in this Pull Request:

Replace sitemap generating code.
#### Testing instructions:
1. Create a few test posts and upload some images.
2. Activate the sitemaps module of Jetpack.
3. Go to "example.com/news-sitemap.xml".
4. Go to "example.com/sitemap.xml". (It may take a refresh or two the first time.)

Note that sitemap.xml links to two other files; one is the root of a tree of sitemaps for pages, the other is the root of a tree of sitemaps for images. 

These changes allow for arbitrarily large sitemap trees per the [spec](sitemaps.org). To test this functionality without having to create an arbitrarily large site, adjust the constant SITEMAP_MAX_ITEMS in modules/sitemaps.php, (5 is a good value.) This constant determines the maximum number of URLs on a single sitemap, so that if it's set to 5 and we have 6 posts then the tree structure kicks in.

The constant SITEMAP_INTERVAL is the minimum number of seconds between rebuilds; it's currently set to 60 for testing.

We reuse the filter ``jetpack_sitemap_post_type`` to allow plugin authors to include custom post types in the sitemap.


#### Proposed changelog entry for your changes:
---
- [no] Make sure your changes respect [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- [NA] Did you make changes, or create a **new .js file**? If [Gulp](http://gulpjs.com/) is installed on your testing environment, run `gulp js:hint` before to commit your changes. It will allow you to [detect errors in Javascript files](http://jshint.com/about/).
- [NA] Did you create a **new action or filter**? [Add inline documentation](https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/#4-hooks-actions-and-filters) to help others understand how to use the action or the filter.
- [no] Create **[unit tests](https://github.com/Automattic/jetpack/tree/master/tests)** if you can. If you're not familiar with Unit Testing, you can check [this tutorial](https://pippinsplugins.com/series/unit-tests-wordpress-plugins/).
